plugins/circleci/src/api/CircleCIApi.ts

```diff
@@ -14,85 +14,41 @@
  * limitations under the License.
  */
 
+import { createApiRef } from '@backstage/core-plugin-api';
 import {
-  CircleCIOptions,
-  getMe,
-  getBuildSummaries,
-  getFullBuild,
-  postBuildActions,
-  BuildAction,
-  BuildWithSteps,
-  BuildStepAction,
-  BuildSummary,
-  GitType,
-} from 'circleci-api';
-import { createApiRef, DiscoveryApi } from '@backstage/core-plugin-api';
-
-/** @public */
-export { GitType };
-
-/** @public */
-export type { BuildWithSteps, BuildStepAction, BuildSummary };
+  Build,
+  PipelineListResponse,
+  RerunWorkflowResponse,
+  Workflow,
+  WorkflowJobListResponse,
+  WorkflowListResponse,
+} from '../types';
+
+export interface CircleCIApi {
+  getPipelinesForProject(
+    projectSlug: string,
+    pageToken?: string,
+  ): Promise<PipelineListResponse>;
+  getWorkflowsForPipeline(
+    pipelineId: string,
+    pageToken?: string,
+  ): Promise<WorkflowListResponse>;
+  getWorkflow(workflowId: string): Promise<Workflow>;
+  getWorkflowJobs(
+    workflowId: string,
+    pageToken?: string,
+  ): Promise<WorkflowJobListResponse>;
+  getBuild(projectSlug: string, buildNumber: number): Promise<Build>;
+  getStepOutput(
```

Comment:
a general note on API design: it's often useful, especially in public surfaces, to only have a single argument that's an object with named fields, instead of lists of arguments that are easy to mix up. So,

```ts
getStepOutput(options: {
  projectSlug: string;
  buildNumber: number;
  ... etc
```

Reply:
Fixed